lib/deaf_grandma.rb

```diff
@@ -21,7 +21,14 @@ def run!
   def speak(input)
 
     #Implement your code here <<<<<<<<<
+    @bye_counter += 1  if (input.include? "BYE")
+    return "SEE YOU LATER SONNY!" if ( @bye_counter >= 3 )
 
+    if (input.upcase == input)
+      return "NOT SINCE 1964!"
```

Comment:
Oh wow, I didn't even pick up on this when I looked at it last night. Yeah, the only explicit return you really need in your code is the one on line 25 where you're checking to see if the bye_counter is >= 3. Otherwise, when your code hits the if, it won't even evaluate the else.